src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,43 +1,65 @@
 import React from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 
-export const TodoModal: React.FC = () => {
-  return (
-    <div className="modal is-active" data-cy="modal">
-      <div className="modal-background" />
-
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+interface Props {
+  todo: Todo;
+  user: User | null;
+  isLoading: boolean;
+  onClose: () => void;
+}
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button type="button" className="delete" data-cy="modal-close" />
-          </header>
+export const TodoModal: React.FC<Props> = ({ todo, user, isLoading, onClose }) => (
+  <div className="modal is-active" data-cy="modal">
+    <div className="modal-background" />
 
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
+    {isLoading ? (
+      <Loader />
+    ) : (
+      <div className="modal-card">
+        <header className="modal-card-head">
+          <div
+            className="modal-card-title has-text-weight-medium"
+            data-cy="modal-header"
+          >
+            Todo #{todo.id}
+          </div>
 
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
+          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+          <button
+            type="button"
+            className="delete"
+            data-cy="modal-close"
+            onClick={onClose}
+            aria-label="Close modal"
+          />
+        </header>
 
-              {' by '}
+        <div className="modal-card-body">
+          <p className="block" data-cy="modal-title">
+            {todo.title}
+          </p>
 
-              <a href="mailto:Sincere@april.biz">Leanne Graham</a>
-            </p>
-          </div>
+          <p className="block" data-cy="modal-user">
+            <strong
+              className={
+                todo.completed ? 'has-text-success' : 'has-text-danger'
+              }
+            >
+              {todo.completed ? 'Done' : 'Planned'}
+            </strong>
+
+            {' by '}
+
+            {user ? (
+              <a href={`mailto:${user.email}`}>{user.name}</a>
+            ) : (
+              'Loading user...'
+            )}
```

Comment:
Consider providing user feedback if the user data fails to load, rather than just displaying 'Loading user...'. This could be a message indicating an error or retry option.